### ISSUE TITLE
m2l_export.loop2gempy

### ISSUE THREAD

**Issue by bluetyson** [Loop3D/map2loop#11]
m2l_export.loop2gempy(test_data_name,tmp_path,vtk_path,output_path+'orientations_clean.csv',                                     output_path+'contacts_clean.csv',tmp_path+'groups_clean.csv',                                     bbox,model_base, model_top,vtk,dtm_reproj_file)  

I pulled the latest one yesterday and getting an error now

The vertical exageration is:  145.18053310360867
Active grids: ['regular']
Cropped raster to geo_model.grid.extent.
depending on the size of the raster, this can take a while...
storing converted file...
Active grids: ['regular' 'topography']
Setting kriging parameters to their default values.
Compiling theano function...
Level of Optimization:  fast_run
Device:  cpu
Precision:  float64
Number of faults:  0
Compilation Done!
Kriging values: 
                                                                                                                                                   values
range                                                                                                                                        2.26307e+06
$C_o$                                                                                                                                        1.21941e+11
drift equations  [3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3]
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
scan_perform.pyx in theano.scan_module.scan_perform.perform()

ValueError: Input dimension mis-match. (input[0].shape[1] = 0, input[1].shape[1] = 5)

During handling of the above exception, another exception occurred:

**Comment by bluetyson** [comment]
My hacked version from earlier in the week is still working though:

if(workflow['model_engine']=='gempy'):

    t1 = time.time()
    #import importlib
    #importlib.reload(m2l_export)


    vtk=False
    richard_loop2gempy(test_data_name,tmp_path,vtk_path,output_path+'orientations_clean.csv',
                                    output_path+'contacts_clean.csv',tmp_path+'groups_clean.csv',
                                    bbox,model_base, model_top,vtk,dtm_reproj_file)

##########################################################################
def richard_loop2gempy(test_data_name: str, tmp_path: str, vtk_path: str, orientations_file: str,
               contacts_file: str, groups_file:str,
               bbox: tuple, model_base: float, model_top: float, vtk: bool, dtm_reproj_file:str = None,
               va=None,
               verbose: bool = False, compute: bool = True):
    """

    :param test_data_name:
    :param tmp_path:
    :param vtk_path:
    :param orientations_file:
    :param contacts_file:
    :param groups_file:
    :param bbox:
    :param model_base:
    :param model_top:
    :param vtk:
    :param dtm_reproj_file:
    :param va: vertical anisotropy. Factor by which all Z coordinates are multiplied by
    :param verbose:
    :param compute:
    :return:
    """
    import gempy as gp
    from gempy import plot
    print("this one running")

    geo_model = gp.create_model(test_data_name)

    # If depth coordinates are much smaller than XY the whole system of equations becomes very unstable. Until
    # I fix it properly in gempy this is a handcrafted hack
    if va is None:
        va = (float(bbox[0]) - float(bbox[2])) / (model_base - model_top)/2

        if va < 3:
            va = 0
        else:
            print('The vertical exageration is: ', va)

    gp.init_data(geo_model, extent=[bbox[0], bbox[2], bbox[1], bbox[3], model_base * va, model_top * va],
                 resolution=(50, 50, 50),
                 path_o=orientations_file,
                 path_i=contacts_file)

    geo_model.modify_surface_points(geo_model.surface_points.df.index, Z=geo_model.surface_points.df['Z'] * va)

    if dtm_reproj_file is not None:
    #if dtm_reproj_file is None:
        # Load reprojected topography to model

        fp = dtm_reproj_file
        geo_model.set_topography(source='gdal', filepath=fp)
        #geo_model.set_topography()
        #geo_model.set_topography(source='random')
		
        print("va", va, "fp",fp)
        print(type(geo_model.grid))
        # Rescaling topography:
        #print("tv b4", geo_model.grid.topography.shape)
        ####geo_model.grid.topography.values[:, 2] *= va
        geo_model._grid.topography.values[:, 2] *= va
        #print("va", geo_model.grid.topography.values)
        ####geo_model.grid.update_grid_values()
        ####geo_model.update_from_grid()
        geo_model._grid.update_grid_values()
        geo_model.update_from_grid()

    # Pile processing:
    print(groups_file)
    contents = np.genfromtxt(groups_file,
                             delimiter=';', dtype='U100')

    # Init dictionary Series:Surfaces
    map_series_to_surfaces = {}
    choice = 0
    for group in contents:
        # Reading surfaces groups
        #surfaces_g = np.atleast_2d(np.genfromtxt(tmp_path + group + '.csv', delimiter=',', dtype='U100'))
        print(group)
        
        surfaces_g = np.atleast_2d(np.genfromtxt(tmp_path + group + '.csv', delimiter=';', dtype='U100'))
        print(surfaces_g)

        # Check if there are several choices
        if surfaces_g.shape[1] > 1:
            surfaces_g = surfaces_g[choice]
        # Deleting the first element since it is not a surface
        surfaces_g = surfaces_g[1:]
        # Creating the mapping dictionary
        map_series_to_surfaces[group] = surfaces_g.tolist()

    if verbose is True:
        print(map_series_to_surfaces)

    # Setting pile to model
    gp.map_series_to_surfaces(geo_model, map_series_to_surfaces, remove_unused_series=False)

    if('Default series' in map_series_to_surfaces):

        #    Removing related data
        del_surfaces = geo_model.surfaces.df.groupby('series').get_group('Default series')['surface']
        geo_model.delete_surfaces(del_surfaces, remove_data=True)

        # Removing series that have not been mapped to any surface
        geo_model.delete_series('Default series')

    if compute is True:
        gp.set_interpolator(geo_model, theano_optimizer='fast_run', dtype='float64')
        gp.compute_model(geo_model)

    # Visualise Model
    p3d = gp.plot_3d(geo_model, plotter_type='background', notebook=False)

    p3d3 = gp.plot_3d(geo_model, notebook=True)
	

    # Save model as vtk
    if vtk:
        vtkexp = gp._plot.export_to_vtk(geo_model, path=vtk_path, name=test_data_name + '.vtk', voxels=False, block=None,
                              surfaces=True)

    return geo_model

**Comment by markjessell** [comment]
As I said in another post I need Miguel or someone close to gempy to take ownership of a map2loop import function, it is too hard to keep up with all the changes!

**Comment by bluetyson** [comment]
Yeah, hopefully in the whole gostin/subsurface thing can work that out.
